src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/DataSource.scala

```diff
@@ -99,6 +100,79 @@ class DataSource(graphFactory: OrientGraphFactory) extends LazyLogging {
 class RawlsTransaction(val graph: OrientGraph, dataSource: DataSource) {
   private val rollbackOnly = new AtomicBoolean(false)
 
+  var readLocks:mutable.Map[Element, Int] = mutable.Map[Element, Int]().withDefaultValue(0)
```

Comment:
I meant to switch this over to `scala.collection.mutable.SynchronizedMap` after the PR comments the last time around. Thanks for the catch.